source/Halibut.Tests/Timeouts/SendingAndReceivingRequestMessagesTimeoutsFixture.cs

```diff
@@ -0,0 +1,179 @@
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut.Diagnostics;
+using Halibut.Tests.Support;
+using Halibut.Tests.Support.TestAttributes;
+using Halibut.Tests.TestServices;
+using Halibut.Tests.Util;
+using Halibut.TestUtils.Contracts;
+using NUnit.Framework;
+using System.Runtime.InteropServices;
+
+namespace Halibut.Tests.Timeouts
+{
+    public class SendingAndReceivingRequestMessagesTimeoutsFixture : BaseTest
+    {
+        [Test]
+        [TestCaseSource(typeof(ServiceConnectionTypesToTest))]
+        public async Task WhenThenNetworkIsPaused_WhileReadingAResponseMessage_ATcpReadTimeoutOccurs_and_FurtherRequestsCanBeMade(ServiceConnectionType serviceConnectionType)
+        {
+            using (var clientAndService = await LatestClientAndLatestServiceBuilder
+                       .ForServiceConnectionType(serviceConnectionType)
+                       .WithPortForwarding(out var portForwarderRef)
+                       .WithEchoService()
+                       .WithDoSomeActionService(() => portForwarderRef.Value.PauseExistingConnections())
+                       .Build(CancellationToken))
+            {
+                portForwarderRef.Value = clientAndService.PortForwarder;
+                var echo = clientAndService.CreateClient<IEchoService>();
+                echo.SayHello("Make a request to make sure the connection is running, and ready. Lets not measure SSL setup cost.");
+
+                var pauseConnections = clientAndService.CreateClient<IDoSomeActionService>(IncreasePollingQueueTimeout());
+
+                var sw = Stopwatch.StartNew();
+                var e = Assert.Throws<HalibutClientException>(() => pauseConnections.Action());
```

Comment:
Is there a way we can make sure this is actually a timeout and something else isn't going wrong? E.g. does the error message give us an indication.

Reply:
It does and we now assert on that